### PR TITLE
Make CounterpartyCommitmentSecrets public

### PR DESCRIPTION
**Pull request by Tibo-lg**
I'm working on implementation and specifications of DLC channels (not on LN for now) and I thought it'd be a good idea to reuse the secret commitment scheme, and thus would like to reuse the `CounterpartyCommitmentSecrets` struct in my code.

I mainly added some comments, let me know if something is off.

### CODE REVIEW

File: lightning/src/ln/chan_utils.rs [hunk 1]
```diff
@@ -173,7 +174,8 @@ impl CounterpartyCommitmentSecrets {
 		48
 	}
 
-	pub(crate) fn get_min_seen_secret(&self) -> u64 {
+	/// Return the minimum index of all stored secrets.
```
[the remark this answers is not in the document]
Comment (Tibo-lg) [reply lightning/src/ln/chan_utils.rs]: Good point added a note (my sentence is maybe a bit awkward if you think of something better let me know).

File: lightning/src/ln/chan_utils.rs [hunk 2]
```diff
@@ -212,8 +217,8 @@ impl CounterpartyCommitmentSecrets {
 		Ok(())
 	}
 
-	/// Can only fail if idx is < get_min_seen_secret
-	pub(crate) fn get_secret(&self, idx: u64) -> Option<[u8; 32]> {
+	/// Returns the secret at index `idx`. Returns `None` if idx is < get_min_seen_secret.
```
[the remark this answers is not in the document]
Comment (Tibo-lg) [reply lightning/src/ln/chan_utils.rs]: Done

File: lightning/src/ln/chan_utils.rs [hunk 3]
```diff
@@ -197,7 +199,10 @@ impl CounterpartyCommitmentSecrets {
 		res
 	}
 
-	pub(crate) fn provide_secret(&mut self, idx: u64, secret: [u8; 32]) -> Result<(), ()> {
+	/// Inserts the secret `secret` at index `idx`. Returns an error if previous
+	/// secrets from the same "bucket" cannot be derived from it. This indicates
```
[the remark this answers is not in the document]
Comment (Tibo-lg) [reply lightning/src/ln/chan_utils.rs]: Sounds better indeed, updated

File: lightning/src/ln/chan_utils.rs [hunk 4]
```diff
@@ -212,8 +217,9 @@ impl CounterpartyCommitmentSecrets {
 		Ok(())
 	}
 
-	/// Can only fail if idx is < get_min_seen_secret
-	pub(crate) fn get_secret(&self, idx: u64) -> Option<[u8; 32]> {
+	/// Returns the secret at index `idx`.
+	/// Returns `None` if idx is < [CounterpartyCommitmentSecrets::get_min_secret].
```
Comment (TheBlueMatt) [inline lightning/src/ln/chan_utils.rs:221]: Note that CI is failing cause this link is bad - its `get_min_seen_secret`.

**Review by TheBlueMatt — COMMENTED** [review]
LGTM aside from the wrong link.

File: lightning/src/ln/chan_utils.rs (hunk 4 above)
Comment (Tibo-lg) [reply lightning/src/ln/chan_utils.rs]: Fixed sorry, I doubled check locally it should be good now 🤞

**Review by TheBlueMatt — DISMISSED** [review]

File: lightning/src/ln/chan_utils.rs [hunk 5]
```diff
@@ -197,7 +200,9 @@ impl CounterpartyCommitmentSecrets {
 		res
 	}
 
-	pub(crate) fn provide_secret(&mut self, idx: u64, secret: [u8; 32]) -> Result<(), ()> {
+	/// Inserts the secret `secret` at index `idx`. Returns Ok(()) if the secret
```
Comment (jkczyz) [inline lightning/src/ln/chan_utils.rs:203]: I think you can leave off the leading non-ticked words "secret" and "index". Likewise elsewhere.

Comment (jkczyz) [inline lightning/src/ln/chan_utils.rs:203]: Add ticks around `Ok(())`

File: lightning/src/ln/chan_utils.rs [hunk 6]
```diff
@@ -212,8 +217,9 @@ impl CounterpartyCommitmentSecrets {
 		Ok(())
 	}
 
-	/// Can only fail if idx is < get_min_seen_secret
-	pub(crate) fn get_secret(&self, idx: u64) -> Option<[u8; 32]> {
+	/// Returns the secret at index `idx`.
+	/// Returns `None` if idx is < [CounterpartyCommitmentSecrets::get_min_seen_secret].
```
Comment (jkczyz) [inline lightning/src/ln/chan_utils.rs:221]: Use ticks around identifiers.

File: lightning/src/ln/chan_utils.rs [hunk 7]
```diff
@@ -173,7 +174,9 @@ impl CounterpartyCommitmentSecrets {
 		48
 	}
 
-	pub(crate) fn get_min_seen_secret(&self) -> u64 {
+	/// Return the minimum index of all stored secrets. Note that indexes start
```
Comment (jkczyz) [inline lightning/src/ln/chan_utils.rs:177]: s/Return/Returns

**Review by jkczyz — COMMENTED** [review]
Just some nits on the doc style

File: lightning/src/ln/chan_utils.rs (hunk 5 above)
Comment (Tibo-lg) [reply lightning/src/ln/chan_utils.rs]: Done I hope I got them all

Comment (Tibo-lg) [reply lightning/src/ln/chan_utils.rs]: Done

File: lightning/src/ln/chan_utils.rs (hunk 6 above)
Comment (Tibo-lg) [reply lightning/src/ln/chan_utils.rs]: Done (I think it was only `idx` on second line missing?)

File: lightning/src/ln/chan_utils.rs [hunk 8]
```diff
@@ -140,10 +140,10 @@ pub fn build_closing_transaction(to_holder_value_sat: u64, to_counterparty_value
 /// Implements the per-commitment secret storage scheme from
 /// [BOLT 3](https://github.com/lightningnetwork/lightning-rfc/blob/dcbf8583976df087c79c3ce0b535311212e6812d/03-transactions.md#efficient-per-commitment-secret-storage).
 ///
-/// Allows us to keep track of all of the revocation secrets of counterarties in just 50*32 bytes
+/// Allows us to keep track of all of the revocation secrets of counter parties in just 50*32 bytes
```
Comment (Tibo-lg) [inline lightning/src/ln/chan_utils.rs:143]: FYI fixed a typo here

Comment (TheBlueMatt) [reply lightning/src/ln/chan_utils.rs:143]: `counterparty` is one word, not two, so I believe `counterparties` would be correct.

Comment (Tibo-lg) [reply lightning/src/ln/chan_utils.rs]: changed, my spellchecker didn't like `counterparties` I guess I'll have to teach him :)

File: lightning/src/ln/chan_utils.rs [hunk 9]
```diff
@@ -140,10 +140,10 @@ pub fn build_closing_transaction(to_holder_value_sat: u64, to_counterparty_value
 /// Implements the per-commitment secret storage scheme from
 /// [BOLT 3](https://github.com/lightningnetwork/lightning-rfc/blob/dcbf8583976df087c79c3ce0b535311212e6812d/03-transactions.md#efficient-per-commitment-secret-storage).
 ///
-/// Allows us to keep track of all of the revocation secrets of counterarties in just 50*32 bytes
+/// Allows us to keep track of all of the revocation secrets of counterparties in just 50*32 bytes
```
Comment (TheBlueMatt) [inline lightning/src/ln/chan_utils.rs:143]: Oh, lol, this shoulnd't be plural anyway. s/counterparties/our counterparty/

**Review by TheBlueMatt — APPROVED** [review]
heh, one more nit, but feel free to ignore it, it doesn't matter that much.

File: lightning/src/ln/chan_utils.rs (hunk 9 above)
Comment (Tibo-lg) [reply lightning/src/ln/chan_utils.rs]: Fixed :)

**Review by TheBlueMatt — DISMISSED** [review]

File: lightning/src/ln/chan_utils.rs (hunk 6 above)
Comment (jkczyz) [reply lightning/src/ln/chan_utils.rs]: Also: [\`CounterpartyCommitmentSecrets::get_min_seen_secret\`]

Comment (Tibo-lg) [reply lightning/src/ln/chan_utils.rs]: Fixed thanks, need to read up more on doc formatting.

**Review by jkczyz — APPROVED** [review]

**Review by TheBlueMatt — APPROVED** [review]

**Pull request merged**